### PR TITLE
Transaction for record ingest

### PR DESCRIPTION
**Pull request by adamdickmeiss**
And handle it being failed, due to cluster value conflict.

### CODE REVIEW

File: server/src/main/java/org/folio/shared/index/storage/Storage.java [hunk 1]
```diff
@@ -179,6 +179,20 @@ Future<Void> deleteBibRecord(SqlConnection conn, String localIdentifier, UUID so
...
+      int iter) {
+
+    return pool.withTransaction(conn ->
+            upsertGlobalRecord(conn, sourceId, globalRecord, matchKeyConfigs)
+        )
+        .recover(x -> {
+          if (iter > 2) {
+            return Future.failedFuture(x);
+          }
+          return upsertGlobalRecord(sourceId, globalRecord, matchKeyConfigs, iter + 1);
```
Comment (skoczko) [inline server/src/main/java/org/folio/shared/index/storage/Storage.java:192]: @adamdickmeiss Why do we have two iterations?

Comment (adamdickmeiss) [reply server/src/main/java/org/folio/shared/index/storage/Storage.java:192]: Because of collision with new cluster value. It's a rare occurrence

Comment (adamdickmeiss) [reply server/src/main/java/org/folio/shared/index/storage/Storage.java:192]: See what was removed in this PR

Comment (skoczko) [reply server/src/main/java/org/folio/shared/index/storage/Storage.java:192]: Is this the Optimistic Locking approach?

**Review by skoczko — APPROVED** [review]

**Pull request merged**